google/cloud/aiplatform/experimental/vertex_model/base.py

```diff
@@ -63,37 +64,80 @@ def f(*args, **kwargs):
 
         training_source = source_utils._make_class_source(obj)
 
-        source = source_utils._make_source(
-            cls_source=training_source,
-            cls_name=cls_name,
-            instance_method=method.__name__,
-        )
-
         with tempfile.TemporaryDirectory() as tmpdirname:
             script_path = pathlib.Path(tmpdirname) / "training_script.py"
 
-            with open(script_path, "w") as f:
-                f.write(source)
+            bound_args = inspect.signature(method).bind(*args, **kwargs)
+            print(type(bound_args))
+
+            # get the mapping of parameter names to types
+            # split the arguments into those that we need to serialize and those that can
+            # be hard coded into the source
+
+            pass_through_params = {}
+            serialized_params = {}
+
+            for parameter_name, parameter in bound_args.arguments.items():
+                if (
+                    bound_args.signature.parameters[parameter_name].kind
+                    is inspect.Parameter.VAR_POSITIONAL
+                ):
+                    parameter_type = type(parameter)
+                    valid_types = [int, float, str] + list(
+                        obj._data_serialization_mapping.keys()
+                    )
+                    if parameter_type not in valid_types:
+                        raise RuntimeError(
+                            f"{parameter_type} not supported. parameter_name = {parameter_name}. The only supported types are {valid_types}"
+                        )
+
+                    if type(parameter) in obj._data_serialization_mapping:
+                        serialized_params[parameter_name] = parameter
+                    else:  # assume primitive
+                        pass_through_params[parameter_name] = parameter
 
             staging_bucket = aiplatform.initializer.global_config.staging_bucket
             if staging_bucket is None:
                 raise RuntimeError(
                     "Staging bucket must be set to run training in cloud mode: `aiplatform.init(staging_bucket='gs://my/staging/bucket')`"
                 )
 
-            obj._training_job = aiplatform.CustomTrainingJob(
-                display_name="my_training_job",
-                script_path=str(script_path),
-                # programatically determine the dependency in the future
-                requirements=["pandas>=1.3"],
-                # https://cloud.google.com/vertex-ai/docs/training/pre-built-containers
-                container_uri="us-docker.pkg.dev/vertex-ai/training/pytorch-xla.1-7:latest",
+            param_name_to_serialized_info = {}
+            for parameter_name, parameter in serialized_params.items():
+                serializer = obj._data_serialization_mapping[type[parameter]][1]
+                parameter_uri = serializer(staging_bucket, parameter, parameter_name)
+
```

Comment:
Possibly add a print or logging statement that the object was serialized. Something like:

`_LOGGER.info(f'{parameter_name} of type {parameter_type} was serialized to {parameter_uri}')`